lib/src/result/error.dart

```diff
@@ -8,19 +8,19 @@ import 'result.dart';
 import 'value.dart';
 
 /// A result representing a thrown error.
-class ErrorResult implements Result<Null> {
+class ErrorResult implements Result<Never> {
   /// The error object that was thrown.
   final Object error;
 
   /// The stack trace corresponding to where [error] was thrown.
-  final StackTrace stackTrace;
+  final StackTrace? stackTrace;
```

Comment:
This should be non-null and we should handle the defaulting if we need to.

Reply:
done